app/frontend/users/UserEdit.res

```diff
@@ -72,7 +74,26 @@ let reducer = (state, action) =>
       currentPassword: currentPassword,
       dirty: true,
     }
-  | UpdateNewPassword(newPassword) => {...state, newPassword: newPassword, dirty: true}
+  | UpdateNewPassword(newPassword) => {
+      let formDictionary = Js.Array2.concat(
+        state.name->Js.String2.split(" "),
+        state.email->Js.String2.split("@"),
+      )
+      let newPasswordAnalysis = if Js.String2.length(newPassword) > 0 {
```

Comment:
We can use `StringUtils.isNotPresent` as a shorter and cleaner abstraction which also checks by trimming the string.